scipy/stats/tests/test_stats.py

```diff
@@ -6615,6 +6615,23 @@ def test_mudholkar_george_equal_fisher_minus_pearson(self):
         # 0.5 here is because logistic = log(u) - log(1-u), i.e. no 2 factors
         assert_approx_equal(0.5 * (Z_f-Z_p), Z, significant=4)
 
+    @pytest.mark.parametrize("variant", ["single", "all"])
+    @pytest.mark.parametrize("method",
+                             ["fisher", "pearson", "tippett",
+                                 "stouffer", "mudholkar_george"])
+    def test_monotony(self, variant, method):
+        # Test that result increases monotonously with respect to input.
+        changing_values = np.linspace(0.1, 0.9, 10)
+        pvalues = np.random.random(7)
```

Comment:
We want reproducible results
```suggestion
        rng = np.random.default_rng()
        pvalues = rng.random(7)
```

Reply:
Would we not have to specify a seed to make this really reproducible? Or does the test environment somehow ensure for a reproducible seed here?

Reply:
We don't rely on global states like `np.random.seed` as this is not recommended anymore. All new code should use explicit seeding of the `np.random.Generator`. Also in this case, a global state would not be reliable as the generated numbers would be different if you launch tests in a different order (which happens all the time). Test must be independent.

Reply:
Sure, but wouldn’t we need to do something like `RNG = np.random.default_rng(42)` then? Because in your suggestion, `default_rng` is globally seeded or even machine-state-seeded, unless I am missing something.

Reply:
Oh yes totally, it's a typo on my side. Fixing the suggestion.